frontend/src/types/api/alerts/def.ts

```diff
@@ -31,6 +33,10 @@ export interface AlertDef {
 	preferredChannels?: string[];
 	broadcastToAll?: boolean;
 	version?: string;
+	// alert v2 fields
+	schemaVersion?: string;
+	notificationSettings?: PostableAlertRuleV2['notificationSettings'];
+	evaluation?: PostableAlertRuleV2['evaluation'];
```

Comment:
Can you help me understand why we are adding these fields here and instead of using the `AlertRuleV2` as the props for the initial definition?

Reply:
Within the new code, we're using `AlertV2` itself.
But if you look at `frontend/src/container/EditRules/index.tsx`, there we are getting the alerts data in the form of `AlertDef`. So I had to update the existing interface.
Then if you look at `CreateAlertV2.tsx`, we're just transforming the `AlertDef` to local state and using it